src/App.tsx

```diff
@@ -1,34 +1,94 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useEffect, useMemo, useState, useCallback,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
-import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { TodoList } from './components/TodoList';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [sortType, setSortType] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos().then(todoFromServer => setTodos(todoFromServer));
+  }, []);
+
+  const filteredAndSortedTodos = useMemo(() => {
+    return todos.filter(todo => {
+      const filteredBySearchQuery = todo.title
+        .toLowerCase()
+        .includes(searchQuery.toLowerCase());
+
+      switch (sortType) {
+        case 'active':
+          return !todo.completed && filteredBySearchQuery;
+
+        case 'completed':
+          return todo.completed && filteredBySearchQuery;
+
+        case 'all':
+        default:
+          return filteredBySearchQuery;
+      }
+    });
+  }, [sortType, todos, searchQuery]);
+
+  const onSearchQueryChange = useCallback(
```

Comment:
```suggestion
  const handleSearchQueryChange = useCallback(
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

Reply:
Oooh I always get confused at this point😅